src/journal.rs

```diff
@@ -130,6 +131,32 @@ impl Journal {
 
         Ok(Some(ret))
     }
+
+    pub fn cursor(&self) -> Result<String> {
+        let mut c_cursor: *mut c_char = ptr::null_mut();
+        let mut cursor: String = "".to_string();
+        if sd_try!(ffi::sd_journal_get_cursor(self.j, &mut c_cursor)) == 0 {
+            unsafe {
+                // Cstr should be used for memory allocated by C
+                cursor = CStr::from_ptr(c_cursor as *const _)
```

Comment:
Should use `CString::from_raw` and not do the copies and manual freeing. (Noted in the other PR, but thought I'd note it here too)